tc/core/polyhedral/schedule_transforms.cc

```diff
@@ -773,5 +774,42 @@ void orderAfter(ScheduleTree* root, ScheduleTree* tree, isl::union_set filter) {
   parent->insertChild(childPos, std::move(seq));
 }
 
+/*
+ * Extract a mapping from the domain elements active at "tree"
+ * to identifiers "ids", where all branches in "tree"
+ * are assumed to have been mapped to these identifiers.
+ * The result lives in a space of the form "tupleId"["ids"...].
+ */
+isl::multi_union_pw_aff extractDomainToIds(
+    const detail::ScheduleTree* tree,
+    const std::vector<mapping::MappingId>& ids,
+    isl::id tupleId = isl::id()) {
```

Comment:
Why did you make the tuple id optional?
I was looking for other uses you have of this function later in this series,
but I can't find any, so maybe I should ask, why is this commit (d60c22af500fe82b1e06efd8197a5708b384a0ce) in this series?

Reply:
I will need it later for promotion.